src/main/java/org/ice4j/util/AsyncQueueHandler.java

```diff
@@ -127,58 +108,22 @@
         @Override
         public void run()
         {
-            final long maxSequentiallyHandledItems =
-                handler.maxSequentiallyHandledItems();
-
-            int sequentiallyHandledItems = 0;
+            long sequentiallyHandledItems = 0;
 
-            while (!cancelled.get())
+            while (running.get())
             {
                 T item;
 
                 synchronized (syncRoot)
                 {
-                    long delay = throttler.getDelayNanos();
-                    if (delay > 0)
-                    {
-                        onDelay(delay, TimeUnit.NANOSECONDS);
-                        return;
-                    }
-
                     if (maxSequentiallyHandledItems > 0 &&
                         sequentiallyHandledItems >= maxSequentiallyHandledItems)
                     {
-                        /*
-                        All instances of AsyncQueueHandler executed on
-                        single shared instance of ExecutorService to better
-                        use existing threads and to reduce number of idle
-                        threads when reader's queue is empty.
-
-                        Having limited number of threads to execute might
-                        lead to other problem, when queue is always
-                        non-empty reader will keep running, while other
-                        readers might suffer from execution starvation. One
-                        way to solve this, is to to artificially interrupt
-                        current reader execution and pump it via internal
-                        executor's queue.
-                        */
                         onYield();
                         return;
                     }
 
-                    try
-                    {
-                        // It is observed that giving very small poll timeout,
-                        // up to hundred nanoseconds, compared to not
-                        // timeout at all has about 5-10% better
-                        // performance in micro-benchmark scenarios.
-                        item = queue.poll(
-                            emptyQueueTimeoutNanoseconds, TimeUnit.NANOSECONDS);
-                    }
-                    catch (InterruptedException e)
-                    {
-                        item = null;
-                    }
+                    item = queue.poll();
```

Comment:
based on your previous comment:

> // It is observed that giving very small poll timeout,	
>                         // up to hundred nanoseconds, compared to not	
>                         // timeout at all has about 5-10% better	
>                         // performance in micro-benchmark scenarios.

did you find that this was no longer the case?

Reply:
> did you find that this was no longer the case?

Honestly saying I was afraid that this trick will not survive review, so I've preemptively removed it to save one cycle of review latency 😄 and I've done that micro-benchmark on Windows, which might give different result than when running on Linux, so that was another reason to remove (I am going to add it back as separate request when I confirm it's benefit on Linux as well, because smaller PR is easier to handle)

Reply:
Ah ok...yeah that sounds good (checking it and doing it in another PR)